pkgs/build-support/node/build-npm-package/hooks/npm-config-hook.sh

```diff
@@ -92,20 +96,7 @@ npmConfigHook() {
 
     local -r lockfileVersion="$(@jq@ .lockfileVersion package-lock.json)"
 
-    if (( lockfileVersion < 2 )); then
-      # This is required because npm consults a hidden lockfile in node_modules to figure out
-      # what to create bin links for. When using an old lockfile offline, this hidden lockfile
-      # contains insufficent data, making npm silently fail to create links. The hidden lockfile
-      # is bypassed when any file in node_modules is newer than it. Thus, we create a file when
-      # using an old lockfile, so bin links work as expected without having to downgrade Node or npm.
-      touch node_modules/.meow
-    fi
-
-    npm rebuild "${npmRebuildFlags[@]}" "${npmFlags[@]}"
-
-    if (( lockfileVersion < 2 )); then
-      rm node_modules/.meow
-    fi
+    npm rebuild $npmRebuildFlags "${npmRebuildFlagsArray[@]}" $npmFlags "${npmFlagsArray[@]}"
```

Comment:
I'll add one when I get more time to work on these PRs tomorrow or Thursday

Reply:
So this was special handling that wasn't needed anymore, so I'm going to remove the `lockfileVersion` variable entirely. The only place that is sensitive to lockfile version would be `prefetch-npm-deps`, which currently panics when it is not lockfile versions 1-3

Since I don't think a comment would make sense here, should a comment be added elsewhere (e.g. in the docs) or should it be considered sufficient that the fetcher/fixup will panic if it encounters a lockfile that's too new?

Reply:
Good enough for me